packages/utils/schema/user-profile.graphql

```diff
@@ -0,0 +1,111 @@
+type ProfileName
+  @createModel(accountRelation: SINGLE, description: "Your name") {
+  name: String! @string(maxLength: 150)
+}
+
+type ProfileAvatar
+  @createModel(
+    accountRelation: SINGLE
+    description: "An image and optional fallback images representing your avatar"
+  ) {
+  original: ImageMetadata!
+  alternatives: [ImageMetadata] @list(maxLength: 3)
+}
+
+type ProfileDescription
+  @createModel(
+    accountRelation: SINGLE
+    description: "A short profile bio / blurb / description"
+  ) {
+  description: String! @string(maxLength: 420)
+}
+
+type ProfileEmoji
+  @createModel(accountRelation: SINGLE, description: "Your spirit emoji") {
+  emoji: String! @string(maxLength: 2)
+}
+
+type ProfileHeaderBackground
+  @createModel(
+    accountRelation: SINGLE
+    description: "An image (3:1 ratio) and optional fallback images to be used as a header background"
+  ) {
+  original: ImageMetadata!
+  alternatives: [ImageMetadata] @list(maxLength: 3)
+}
+
+type ProfileURL
+  @createModel(
+    accountRelation: SINGLE
+    description: "Your home on the internet"
+  ) {
+  url: String! @string(maxLength: 240)
+}
+
+type ProfileHomeLocation
+  @createModel(accountRelation: SINGLE, description: "Where you live") {
+  homeLocation: String! @string(maxLength: 140)
+}
+
+type ProfileUsername
+  @createModel(
+    accountRelation: SINGLE
+    description: "A representation of your identity that can be used in an interface, typically as part of a URL"
+  ) {
+  # pattern: "^[a-z0-9-_]+$"
+  username: String! @string(maxLength: 383)
```

Comment:
I don't know what the limits on DNS name sizes are, but I'm pretty sure they exist and aren't huge.

```suggestion
  username: String! @string(maxLength: 128)
```

Reply:
I don't think we want to make this max smaller since we will porting over old data that has a higher max